stl/inc/vector

```diff
@@ -2882,6 +2882,63 @@ namespace pmr {
     using vector = _STD vector<_Ty, polymorphic_allocator<_Ty>>;
 } // namespace pmr
 #endif // _HAS_CXX17
+
+#if _HAS_IF_CONSTEXPR
+// VARIABLE TEMPLATE _Is_vb_iterator
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_const_iterator<_Alloc>> = true;
+
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_iterator<_Alloc>> = true;
+
+template <class _FwdIt, class _Ty>
+_CONSTEXPR20 void _Fill_vbool(_FwdIt _First, _FwdIt _Last, const _Ty& _Val) {
+    // Set [_First, _Last) to _Val
+    if (_First == _Last) {
+        return;
+    }
+
+    _Vbase* _UFirst            = const_cast<_Vbase*>(_First._Myptr);
+    const _Vbase* const _ULast = _Last._Myptr;
+
+    const auto _FirstSourceMask = static_cast<_Vbase>(-1) << _First._Myoff;
+    const auto _LastSourceMask  = static_cast<_Vbase>(-1) >> (_VBITS - _Last._Myoff);
+    const auto _FirstDestMask   = static_cast<_Vbase>(~_FirstSourceMask);
+    const auto _LastDestMask    = static_cast<_Vbase>(~_LastSourceMask);
+    const auto _FillVal         = static_cast<_Vbase>(_Val ? -1 : 0);
+
+    if (_UFirst == _ULast) {
+        const auto _SourceMask = _FirstSourceMask & _LastSourceMask;
+        const auto _DestMask   = _FirstDestMask | _LastDestMask;
+        *_UFirst               = (*_UFirst & _DestMask) | (_FillVal & _SourceMask);
+        return;
+    }
+
+    *_UFirst = (*_UFirst & _FirstDestMask) | (_FillVal & _FirstSourceMask);
+    ++_UFirst;
+
+    if (true
+#ifdef __cpp_lib_is_constant_evaluated
+        && !_STD is_constant_evaluated()
+#endif // __cpp_lib_is_constant_evaluated
+    ) {

```

Comment:
I hate this but get why it's required, can anyone think of a better thing to do here, because it beats me

Reply:
I guess it depends on support for `__cpp_lib_is_constant_evaluated` by all supported compilers. (Similar to `if constexpr` from C++17)

Reply:
We could do this instead?  
```
#ifdef __cpp_lib_is_constant_evaluated
    if (_STD is_constant_evaluated()) {
        // put current "else" branch code here with for loop
    } else 
#endif
    {
        // put current "if" branch code here with memset
    }
```
if anyone likes that better...

Reply:
That is  much better as it removes the dead code completely.

Honestly I cannot remember why I wrote it that way